connectors/src/connectors/zendesk/temporal/activities.ts

```diff
@@ -381,6 +382,78 @@ export async function syncZendeskArticleBatchActivity({
   return { hasMore: has_more, afterCursor: after_cursor };
 }
 
+export async function syncZendeskTicketsActivityBatch({
+  connectorId,
+  brandId,
+  currentSyncDateMs,
+  forceResync,
+  pageSize,
+  cursor,
+}: {
+  connectorId: ModelId;
+  brandId: number;
+  currentSyncDateMs: number;
+  forceResync: boolean;
+  pageSize: number;
+  cursor: string | null;
+}) {
+  const connector = await _getZendeskConnectorOrRaise(connectorId);
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+  const loggerArgs = {
+    workspaceId: dataSourceConfig.workspaceId,
+    connectorId,
+    provider: "zendesk",
+    dataSourceId: dataSourceConfig.dataSourceId,
+  };
+
+  const { subdomain, accessToken } = await getZendeskSubdomainAndAccessToken(
+    connector.connectionId
+  );
+
+  // Fetch tickets in batches
+  const { tickets, meta } = await fetchZendeskTicketsInBrand({
+    subdomain,
+    accessToken,
+    pageSize,
+    cursor,
+  });
+
+  const zendeskApiClient = createZendeskClient({ subdomain, accessToken });
+  const users = await zendeskApiClient.users.list();
+
+  const res = await concurrentExecutor(
+    tickets,
+    async (ticket) => {
+      const comments = await zendeskApiClient.tickets.getComments(ticket.id);
```

Comment:
you have to change the subdomain of the client before fetching here

Reply:
the subdomain we get with `getZendeskSubdomainAndAccessToken` is the base subdomain for the connection (ergo the client), it does not know about which brand to choose